cmd/cri-containerd/options/options.go

```diff
@@ -84,7 +99,31 @@ func (c *CRIContainerdOptions) AddFlags(fs *pflag.FlagSet) {
 // before flags are accessed by the program. Ths fuction adds flag.CommandLine
 // (the default set of command-line flags, parsed from os.Args) and then calls
 // pflag.Parse().
-func InitFlags() {
-	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
-	pflag.Parse()
+// precedence:  commandline > configfile > default
+func (c *CRIContainerdOptions) InitFlags(fs *pflag.FlagSet) {
+	fs.AddGoFlagSet(flag.CommandLine)
+
+	commandline := os.Args[1:]
+	err := fs.Parse(commandline) //this time:  config = default + commandline(on top)
+	if err != nil {
+		return
+	}
+	err = loadConfigFile(c.ConfigFilePath, &c.Config) //config = default + commandline + configfile(on top)
+	if err != nil {
+		return
+	}
+
+	err = fs.Parse(commandline) //config = default + commandline + configfile + commandline(on top)
```

Comment:
What is the reason for applying the command line twice? (in both L107 and L116).

Reply:
Because the values from command line has the highest priority. So I must get the path of toml configuration file from command line, it trigger the first parse.
The first parse generate the the default value and the value from command line  at the same time.
But the priority of toml config value is more higher than of default value, So I have not another way to insert toml config value between default value and command line value. So I trigger twice parses, one for default value, one for commandline value.

Reply:
Sorry, I missed the notification of your reply.

You explanation is completely reasonable, but it is rather rather subtle so IMHO it would be worth writing a paragraph in a code comment on why it parses the command line twice (something based on your comment would be good).

Reply:
I agree with you. I will do it.